pkg/cmd/render/render.go

```diff
@@ -109,98 +134,218 @@ func runRenderCmd(cmd *cobra.Command, args []string) {
 	log.SetLevel(level)
 	log.Debug("debug logging enabled")
 
-	operatorDisabled := isDisabled()
+	err = render()
+	if err != nil {
+		log.WithError(err).Fatal("failed while rendering bootstrap assets")
+	}
+}
+
+func render() error {
+	operatorDisabledViaConfigmap := isDisabledViaConfigmap()
+
+	installConfigMode, err := getModeFromInstallConfig()
+	if err != nil {
+		return err
+	}
+	if !isValidMode(installConfigMode) {
+		return fmt.Errorf("invalid mode defined: %s", installConfigMode)
+	}
+
+	installConfigModeManual := installConfigMode == string(operatorv1.CloudCredentialsModeManual)
+
+	// Make sure deprecated configmap setting doesn't conflict with install-config contents.
+	// Otherwise there could be a race between the configmap and the operator config
+	// resources landing causing unpredictable runtime behavior of the CCO pod.
+	if operatorDisabledViaConfigmap && installConfigMode != "" {
+		if operatorDisabledViaConfigmap != installConfigModeManual {
```

Comment:
```suggestion
		if !installConfigModeManual {
```
Or don't even define `installConfigModeManual`
```suggestion
		if installConfigMode == string(operatorv1.CloudCredentialsModeManual) {
```